src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -122,9 +116,115 @@ private string[] usedObervedDataFrom(ModelSimulation simulation)
          return null;
       }
 
-      public override Task<ModelSimulation> MapToModel(SnapshotSimulation snapshot, PKSimProject project)
+      public override async Task<ModelSimulation> MapToModel(SnapshotSimulation snapshot, PKSimProject project)
+      {
+         var simulation = await createSimulationFrom(snapshot, project);
+
+         simulation.Solver = await _solverSettingsMapper.MapToModel(snapshot.Solver);
+         simulation.OutputSchema = await _outputSchemaMapper.MapToModel(snapshot.OutputSchema);
+         simulation.OutputSelections = await _outputSelectionsMapper.MapToModel(snapshot.OutputSelections, simulation);
+
+         await updateParameters(simulation, snapshot.Parameters);
+         await updateAdvancedParameters(simulation, snapshot.AdvancedParameters);
+
+         updateUsedObservedData(simulation, snapshot.ObservedData, project);
+
+         return simulation;
+      }
+
+      private async Task<ModelSimulation> createSimulationFrom(SnapshotSimulation snapshot, PKSimProject project)
+      {
+         var simulationSubject = simulationSubjectFrom(snapshot, project);
+         var compounds = compoundsFrom(snapshot.Compounds, project);
+         var modelProperties = modelPropertiesFrom(snapshot.Model, simulationSubject);
+
+         var simulation = _simulationFactory.CreateFrom(simulationSubject, compounds, modelProperties);
+         MapSnapshotPropertiesToModel(snapshot, simulation);
+
+         await mapCompoundProperties(snapshot.Compounds, project, simulation);
+         simulation.EventProperties = await _eventPropertiesMapper.MapToModel(snapshot.Events, project);
+
+         //Once all used building blocks have been set, we need to ensure that they are also synchronized in the in the simulation
```

Comment:
Change that comment to remove the duplicated 'in the in the'

Reply:
done